rest/src/test/java/eu/openg/guessnumberapi/rest/route/GetGamesRouteTest.java

```diff
@@ -0,0 +1,92 @@
+import eu.openg.guessnumberapi.rest.entity.converter.RestResponseConverter;
+import eu.openg.guessnumberapi.rest.exception.ServerErrorException;
+import eu.openg.guessnumberapi.usecase.api.BoundaryGame;
+import eu.openg.guessnumberapi.usecase.api.GetGamesUseCase;
+import eu.openg.guessnumberapi.usecase.api.UseCaseFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import spark.Request;
+import spark.Response;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class GetGamesRouteTest {
+    private static GetGamesRoute getGamesRoute;
+    @Mock
+    private static UseCaseFactory useCaseFactory;
+    @Mock
+    private static JSONSerializer serializer;
+    @Mock
+    private static GetGamesUseCase getGamesUseCase;
+    @Mock
+    private static Request request;
+    @Mock
+    private static Response response;
+    @Mock
+    private static RestResponseConverter restResponseConverter;
+
+    @BeforeEach
+    void setUp() {
+        getGamesRoute = new GetGamesRoute(useCaseFactory, serializer, restResponseConverter);
+    }
+
+    @Test
+    void returnGamesWhenDbIsNotEmpty() {
+        int gamesCount = 4;
+        String responseBody = assignResponseBody(gamesCount);
+        List<RestGame> restGames = returnRestGames(gamesCount);
+        List<BoundaryGame> boundaryGames = returnBoundaryGames(gamesCount);
+        initMocks(restGames, boundaryGames, responseBody);
+        getGamesRoute.handle(request, response);
+        verify(response).body(responseBody);
+    }
+
+    @Test
+    void returnGamesWhenDbIsEmpty() {
+        when(useCaseFactory.buildGetGamesUseCase()).thenReturn(getGamesUseCase);
+        assertThrows(ServerErrorException.class, () -> getGamesRoute.handle(request, response));
+    }
+
+    private List<BoundaryGame> returnBoundaryGames(int gamesCount) {
+        return IntStream.rangeClosed(1, gamesCount)
+                .mapToObj(i -> new BoundaryGame(i, i, i))
+                .collect(Collectors.toList());
+    }
+
+    private List<RestGame> returnRestGames(int gamesCount) {
+        return IntStream.rangeClosed(1, gamesCount)
+                .mapToObj(i -> new RestGame(i, i, i))
+                .collect(Collectors.toList());
+    }
+
+    private String assignResponseBody(int gamesCount) {
+        String responseBody = IntStream.rangeClosed(1, gamesCount)
+                .mapToObj(i -> "{\"gameId\":" + i + ",\"guessCount\":" + i + ",\"actualNumber\":" + i + "},")
+                .collect(Collectors.joining());
```

Comment:
beveik normaliai padarei. tam ilgam stringe nereikia gale ",", o Collectors.joining() priima vieną paramą - delimiter, tai paduok "," ir td nereiks tau nuiminėt pačiam to kabliuko gale